src/uniswapMappings/uniswapPools.ts

```diff
@@ -0,0 +1,143 @@
+  token0.save()
+  token1.save()
+}
+
+export function handleMint(event: MintEvent): void {
+  let poolAddress = event.address.toHexString()
+  let pool = UniswapPool.load(poolAddress)
+
+  let token0 = Token.load(pool.token0)
+  let token1 = Token.load(pool.token1)
+  let amount0 = convertTokenToDecimal(event.params.amount0, BigInt.fromI32(token0.decimals))
+  let amount1 = convertTokenToDecimal(event.params.amount1, BigInt.fromI32(token1.decimals))
+
+  // Pools liquidity tracks the currently active liquidity given pools current tick.
+  // We only want to update it on mint if the new position includes the current tick.
+  if (
+    pool.tick !== null &&
+    BigInt.fromI32(event.params.tickLower).le(pool.tick as BigInt) &&
+    BigInt.fromI32(event.params.tickUpper).gt(pool.tick as BigInt)
+  ) {
+    pool.liquidity = pool.liquidity.plus(event.params.amount)
+  }
+
+  pool.totalValueLockedToken0 = pool.totalValueLockedToken0.plus(amount0)
+  pool.totalValueLockedToken1 = pool.totalValueLockedToken1.plus(amount1)
+
+  // TODO: Update Tick's volume, fees, and liquidity provider count. Computing these on the tick
+  // level requires reimplementing some of the swapping code from v3-core.
+
+  token0.save()
+  token1.save()
+  pool.save()
+
+}
+
+export function handleBurn(event: BurnEvent): void {
+  let poolAddress = event.address.toHexString()
+  let pool = UniswapPool.load(poolAddress)
+
+  let token0 = Token.load(pool.token0)
+  let token1 = Token.load(pool.token1)
+  let amount0 = convertTokenToDecimal(event.params.amount0, BigInt.fromI32(token0.decimals))
+  let amount1 = convertTokenToDecimal(event.params.amount1, BigInt.fromI32(token1.decimals))
+
+  // Pools liquidity tracks the currently active liquidity given pools current tick.
+  // We only want to update it on burn if the position being burnt includes the current tick.
+  if (
+    pool.tick !== null &&
+    BigInt.fromI32(event.params.tickLower).le(pool.tick as BigInt) &&
+    BigInt.fromI32(event.params.tickUpper).gt(pool.tick as BigInt)
+  ) {
+    pool.liquidity = pool.liquidity.minus(event.params.amount)
+  }
+
+  pool.totalValueLockedToken0 = pool.totalValueLockedToken0.minus(amount0)
+  pool.totalValueLockedToken1 = pool.totalValueLockedToken1.minus(amount1)
+
+  token0.save()
+  token1.save()
+  pool.save()
+}
+
+export function handleSwap(event: SwapEvent): void {
+  let bundle = Bundle.load('1')
+  let pool = UniswapPool.load(event.address.toHexString())
+
+  // hot fix for bad pricing
+  if (pool.id == '0x9663f2ca0454accad3e094448ea6f77443880454') {
+    return
+  }
+
+  let token0 = Token.load(pool.token0)
+  let token1 = Token.load(pool.token1)
+
+  // amounts - 0/1 are token deltas: can be positive or negative
+  let amount0 = convertTokenToDecimal(event.params.amount0, BigInt.fromI32(token0.decimals))
+  let amount1 = convertTokenToDecimal(event.params.amount1, BigInt.fromI32(token1.decimals))
+
+  // need absolute amounts for volume
```

Comment:
better write the comments in imperative for. i.e. Get absolue amounts for volume calculations

Reply:
ok, will change that.